orttraining/orttraining/core/agent/training_agent.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <thread>
+#include <future>
+
+#include "core/common/common.h"
+#include "core/common/logging/logging.h"
+#include "core/framework/framework_common.h"
+#include "core/session/IOBinding.h"
+#include "core/session/inference_session.h"
+#include "orttraining/training_ops/cpu/controlflow/ort_tasks.h"
+
+
+namespace onnxruntime {
+namespace training {
+class IOBinding;
+
+class TrainingAgent {
+
+  public:
+    explicit TrainingAgent(InferenceSession* session);
+    virtual ~TrainingAgent();
```

Comment:
>virtual [](start = 4, length = 7)

maybe i missed it, but i didn't see any derived classes. do we need virtual methods?

Reply:
indeed, removed.